components/sensor/max17043.rst

```diff
@@ -0,0 +1,104 @@
+Unlike traditional fuel gauges, the ModelGauge algorithm eliminates the need for battery relearn cycles
+and an external current-sense resistor.
+
+In low power applications, it's very important to report battery levels accurately. By utilising ESPHome's ``deep_sleep``
+component together with a MAX17043, projects can run for extended periods and the user can be confident of the amount of battery remaining.
+This overcomes the reality that measured battery voltage does not correlate well at all with remaining battery charge.
+
+.. note::
+
+    See hardware design discussion below - it's important to leave the MAX17043 powered on during deep sleep.
+
+.. figure:: images/max17043.jpg
+    :align: center
+    :width: 60.0%
+
+    DFRobot DFR0563 Gravity I2C 3.7V Li battery fuel gauge.
+
+.. _DFRobot: https://www.dfrobot.com/product-1734.html
+
+Configuration
+*************
+
+.. code-block:: yaml
+
+    # Example configuration entry
+    sensor:
+      - platform: max17043
+        id: max17043_id
+        i2c_id: i2c_max17043
+        battery_voltage:
+          name: "Battery Voltage"
+        battery_level:
+          name: "Battery"
+
+Sensors
+*******
+
+- **battery_voltage** (*Optional*, float): The voltage measured at the LIPO battery.
+- **battery_level** (*Optional*, float): The percentage of battery remaining using the ModelGauge™ battery-modelling scheme.
+
+Hardware design considerations
+******************************
+
+It's important to realise that the relationship between battery voltage and remaining battery level
+is poorly correlated as well as being non-linear.
+
+The MAX17043 works by continually monitoring charge and discharge to assess how much battery capacity remains.
+When first powered on, it makes an assumption that the measured voltage has been in a relaxed state for
+30 minutes. This best first guess does not have a lasting impact because it monitors relative state-of-charge
+over time.
+
+Deep sleep cycles are most often designed so the device wakes up for short
+periods to do its business and then sleeps for a much longer period. It's critical to allow the MAX17043 to maintain
+state during the sleep phase.
+
+If power is removed from the MAX17043 then each time the ESP comes out of deep sleep the MAX17043 will
+have to start again from a new best guess.
+It will not be able to use past charge and discharge behaviour to work its magic -
+significantly diminishing the point of using a MAX17043.
+
+Current consumption during device sleep mode is extremely low (maximum of 3µA compared to up to 75µA maximum when active).
+The driver enables the user to put the device to sleep just before deep sleep commences using the `sleep_mode` action.
+When the ESP comes out of deep sleep the driver clears the MAX17043 sleep bit.
+
+For example,
+
+.. code-block:: yaml
+
+    on_...:
+      then:
+        - lambda: |-
+            id(max17043_id).sleep_mode();
```

Comment:
This example should use the ESPHome action to do this (but see https://github.com/esphome/esphome/pull/7522#discussion_r1796606434).